src/moonlink/src/storage/iceberg/rest_catalog_test.rs

```diff
@@ -0,0 +1,83 @@
+#[cfg(feature = "rest-catalog")]
+#[cfg(test)]
+mod tests {
+    use crate::storage::iceberg::rest_catalog::RestCatalog;
+    use crate::storage::iceberg::rest_catalog_test_guard::RestCatalogTestGuard;
+    use crate::storage::iceberg::rest_catalog_test_utils::*;
+    use iceberg::{Catalog, NamespaceIdent};
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_create_table() {
+        let namespace = random_string();
+        let table = random_string();
+        let _guard = RestCatalogTestGuard::new(namespace.clone(), None)
+            .await
+            .expect("Namespace creation fail");
+        let config = default_rest_catalog_config();
+        let catalog = RestCatalog::new(config).await;
+        let namespace = NamespaceIdent::new(namespace);
+        let table_creation = default_table_creation(table);
+        catalog
+            .create_table(&namespace, table_creation)
+            .await
+            .expect("Table creation fail");
```

Comment:
nit: I won't be too picky here, but if you want to do proper error handling with `expect`, might be better to include namespace and table name here; because we could still debug after test failure with local rest catalog container.

Same for all below.